config/endpoint/endpoint.go

```diff
@@ -327,11 +328,28 @@ func (e *Endpoint) EvaluateHealth() *Result {
 }
 
 func (e *Endpoint) getIP(result *Result) {
-	if ips, err := net.LookupIP(result.Hostname); err != nil {
+
+	resolver := net.DefaultResolver
+
+	if e.ClientConfig.HasCustomDNSResolver() {
+		dnsResolver := e.ClientConfig.DNSResolverConfig
+		resolver = &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{}
+				return d.DialContext(ctx, dnsResolver.Protocol, dnsResolver.Host+":"+dnsResolver.Port)
+			},
+		}
+	}
+
```

Comment:
can you remove the unnecessary empty newlines & replace them with comments that explains what's going on?